pkg/repository/v1_repository.go

```diff
@@ -379,6 +380,23 @@ func (r *V1Repository) updateComponentManifest(id string) (*v1manifest.Component
 	return &component, nil
 }
 
+// DownloadComponent downloads a component with specific version from repository
+func (r *V1Repository) DownloadComponent(
```

Comment:
What does this function do differently to `UpdateComponent`?

Reply:
I an not so quite sure, but it should be the same as Repository.DownloadComponent.
and here version can only be a not empty version, in UpdateComponent seems it will choose a version if it's empty and may update the version.Here just download and install a specified version

Reply:
If version is supplied in the component spec, then UpdateComponent will install that version.

Reply:
I will keep this now as v0 in this pr since it's just a simple func. not sure will it be nicer to replace this by UpdateComponent